src/Microsoft.AspNetCore.Hosting/Internal/HostingApplication.cs

```diff
@@ -64,21 +75,68 @@ public Context CreateContext(IFeatureCollection contextFeatures)
                 // Non-inline
                 LogRequestStarting(httpContext);
             }
-
             if (diagnoticsEnabled)
             {
+#if !NET451
+                context.Activity = StartActivity(httpContext);
+#endif
                 // Non-inline
                 RecordBeginRequestDiagnostics(httpContext, startTimestamp);
             }
 
-            // Create and return the request Context
-            return new Context
+            // Fill and return the request Context
+            context.HttpContext = httpContext;
+            context.Scope = scope;
+            context.StartTimestamp = startTimestamp;
+            return context;
+        }
+
+#if !NET451
+        private Activity StartActivity(HttpContext httpContext)
+        {
+            var activity = new Activity(ActivityName);
+            StringValues requestId;
+            if (httpContext.Request.Headers.TryGetValue(_activityTrackingOptions.RequestIdHeaderName, out requestId))
+            {
+                try
+                {
+                    activity.SetParentId(requestId);
+                }
+                catch (ArgumentException ex)
```

Comment:
The try-catch should cause the Write() call but it can skip anything associated with setting up the activity.  It can even set Activity.Current.     I am just trying to simplify the code (less error handling more real logic).  

However thinking about it, I think I would prefer if Activity never throws (DiagnosticSource never throws today (assuming its target does not)) and that is good.   If I look at when Activity throws, it is mostly things that could be contracts (programmer errors, not runtime conditions).   We should probably have the ability to register a 'error callback' on Activity, and then otherwise never fail (simply do the best you can).     This allows us to get rid of try-catch 'goo' like this completely.   I will look into it.